### PR TITLE
Moving backward compat authrole code in one place and also removed redundant fields

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
With the introduction of the default values in service account 
https://github.com/flyteorg/flyteadmin/blob/6e03bd4ac2ad3bb950e8fa1ed64f4bc5c6058bcd/pkg/runtime/application_config_provider.go#L34

And this logic 
https://github.com/flyteorg/flyteadmin/blob/6e03bd4ac2ad3bb950e8fa1ed64f4bc5c6058bcd/pkg/manager/impl/execution_manager.go#L788

Hence even if authRole  which is deprecated field was set in launchplan or execution request, it was being ignored
since this logic was being resolved after the application default values were added.

Also as part of this PR cleaned up redundant fields in ExecutionParameters for Security Context and fetching those from 

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/execution_manager.go [hunk 1]
```diff
@@ -511,6 +511,14 @@ func (m *ExecutionManager) getExecutionConfig(ctx context.Context, request *admi
 		if launchPlan.Spec.WorkflowId != nil {
 			workflowName = launchPlan.Spec.WorkflowId.Name
 		}
+		// Backward compatibility changes to get security context from auth role.
```
Comment (katrogan) [inline pkg/manager/impl/execution_manager.go:514]: this is getting so hairy :( can you add a more detailed comment why this is necessary?

**Review by katrogan — DISMISSED** [review]

**Comment by convexquad** [comment]
@pmahindrakar-oss if it is possible for you to finish this PR and commit it soon, that would be great! I need this fix in order to complete the upgrade of my Flyte backend system (i.e. Flyte Admin, Propeller, etc).

However, I would rather have this fix be made correctly the first time - if you need more time to keep working on it or to keep testing, I would rather you do that than try to rush the fix.

**Comment by convexquad** [comment]
@katrogan could you take this the rest of the way if for some reason @pmahindrakar-oss is not available to finish it?

**Comment by pmahindrakar-oss** [comment]
@convexquad would update on this today.  Unit tests do cover a lot of the older code too and i will try simulate workflows in older deprecated fields.
Also would be helpful if your team starts migrating your older workflows to new security context as that is the model going forward. We don't have a post deprecation strategy yet but wanted to get your opinion to avoid such issues in the future.

Having deprecated code across releases increases the test matrix and higher maintainability cost. 
Would prefer to remove deprecated field after 3 monthly releases and communication in the release notes for the same.

So a field marked as deprecated would be mentioned in the release and post 3 releases this field would be removed.
This would help remove a lot of tech debt to be carried forward.

File: pkg/manager/impl/execution_manager.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply pkg/manager/impl/execution_manager.go]: Hi @katrogan added comments for these

**Comment by pmahindrakar-oss** [comment]
I have tested this my changing flytectl code to send older authRole during registration of workflows and verified that the same authRole is being picked by the workflows during execution in both regular and scheduled workflow case.
@katrogan @convexquad please review when you get a chance.

**Review by katrogan — APPROVED** [review]

**Review by convexquad — COMMENTED** [review]
👍 I think this change will work correctly for scheduled launch plans - they should pick up the correct k8s service account with this change.

However @pmahindrakar-oss just to be clear, this is technically not a backwards compatibility issue. Without this change, scheduled launch plans will only run with the "default" k8s service account even with the most recent flytectl, flytekit, Flyte Admin, etc. versions.

**Comment by convexquad** [comment]
@katrogan could you merge this PR and release a new Flyte Admin version for it, I guess. I can put it through the checks in my test cluster so we should catch any secondary issues quickly at least.

**Comment by katrogan** [comment]
hey @convexquad if you'd like I've built an image with these changes: ghcr.io/flyteorg/flyteadmin:v1.1.20-security-ctx that you can test with. that way we can catch any outstanding issues before merging this change in

**Comment by convexquad** [comment]
> hey @convexquad if you'd like I've built an image with these changes: ghcr.io/flyteorg/flyteadmin:v1.1.20-security-ctx that you can test with. that way we can catch any outstanding issues before merging this change in

@katrogan that works, I will post a comment on this PR or in the Flyte Slack workspace once I am done testing

**Comment by katrogan** [comment]
thank you @convexquad for taking the time to test out this change (and hopefully it works!🤞)

**Comment by convexquad** [comment]
@katrogan I tested launching workflows with a scheduled launch plan; with `flytectl create execution` and from Flyte Console and I got the correct declared k8s service account in all three tests. Could you merge this and publish a release?

**Review by katrogan — APPROVED** [review]

**Comment by pmahindrakar-oss** [comment]
@convexquad Most recent flytectl and flytekit and any newly registered launchplan scheduled or unscheduled will be registered with new security context field. This change will take care of any older registered launchplans which had authRole in them.

Thanks for checking on this testing out.
